projects/com.oracle.truffle.llvm.tools/src/com/oracle/truffle/llvm/tools/GCC.java

```diff
@@ -60,9 +61,23 @@ public static void compileToLLVMIR(File toBeCompiled, File destinationFile) {
         } else {
             throw new AssertionError(toBeCompiled);
         }
-        String[] command = new String[]{tool, "-I " + LLVMBaseOptionFacade.getProjectRoot() + "/../include", "-S", dragonEggOption(), "-fplugin-arg-dragonegg-emit-ir", "-o " + destinationFile,
-                        toBeCompiled.getAbsolutePath()};
-        ProcessUtil.executeNativeCommandZeroReturn(command);
+        if (destinationFile.getName().endsWith(".bc")) {
+            // TODO: Check for more elegant way to generate .bc file for FORTRAN source file
+
+            // Creating .ll file for the FORTRAN source file
+            String llFileGenerationcommand = tool + " -I " + LLVMBaseOptionFacade.getProjectRoot() + "/../include -S " + dragonEggOption() + " -fplugin-arg-dragonegg-emit-ir  -o /tmp/temp.ll " +
+                            toBeCompiled.getAbsolutePath();
+            ProcessUtil.executeNativeCommandZeroReturn(llFileGenerationcommand);
+
+            // Converting .ll file to .bc file
+            String llToBCConversionCommand = LLVM_AS_PATH.toString() + " /tmp/temp.ll -o " + destinationFile;
```

Comment:
Could you add the functionality for using `llvm-as` in the `LLVMAssembler` class, and remove it from here? Also, you can reference the `llvm-as` tool path by looking it up from the `LLVMToolPaths` class.